shark_turbine/kernel/wave/scheduling/graph_utils.py

```diff
@@ -0,0 +1,132 @@
+import torch.fx as fx
+from random import shuffle, seed
+from collections import defaultdict
+
+
+def find_strongly_connected_components(
+    graph: fx.Graph, random_seed: int
+) -> dict[fx.Node, list[fx.Node]]:
+    """
+    Find the strongly connected components in the graph.
+    Returns a list of strongly connected components.
```

Comment:
Some links to the actual algorithm will help, I assume you are doing https://en.wikipedia.org/wiki/Kosaraju%27s_algorithm, sorting by `node.f` instead of maintaining explicit nodes stack.